bitcoin/src/pow.rs

```diff
@@ -234,6 +234,20 @@ impl Target {
     /// [`difficulty`]: Target::difficulty
     #[cfg_attr(all(test, mutate), mutate)]
     pub fn difficulty_float(&self) -> f64 { TARGET_MAX_F64 / self.0.to_f64() }
+
+    /// Computes the minimum valid [`Target`] threshold allowed for a block in which a difficulty
+    /// adjustment occurs.
+    ///
+    /// The difficulty can only decrease or increase by a factor of 4 max on each difficulty
+    /// adjustment period.
+    pub fn min_difficulty_transition_threshold(&self) -> Self { Self(self.0 >> 2) }
+
+    /// Computes the maximum valid [`Target`] threshold allowed for a block in which a difficulty
+    /// adjustment occurs.
+    ///
+    /// The difficulty can only decrease or increase by a factor of 4 max on each difficulty
+    /// adjustment period.
+    pub fn max_difficulty_transition_threshold(&self) -> Self { Self(self.0 << 2) }
```

Comment:
This will debug-panic on large values, is this OK/correct?

Reply:
What makes you say that @Kixunil ? We use `wrapping_shl` to implement `<<` so I thought this couldn't panic, what am I missing?

@wpaulino why did the check against `MAX` get removed here? In the previous iteration I only meant to comment that I didn't like the check in the `min_difficulty_transition_threshold` method, in the max one I thought it was correct.

Reply:
Oh yes, sorry, my bad for not remembering all the context when reviewing. I did read that before.

Reply:
Oh, sorry I got confused, it'd only panic if the argument was too large, 2 is not too large.

Reply:
I constantly have this same confusion about left-shift. It's a little weird, this notion of "overflow" is different from that for the arithmetic opcodes.